sources/core/FastIsType.h

```diff
@@ -31,7 +31,7 @@ inline Typeto* CastTo(TypeObjectFrom* ptr)
     }
     if (FastIsType<Typeto>(ptr->RT_ITEM()))
     {
-        return static_cast<Typeto*>(ptr);
+        return reinterpret_cast<Typeto*>(ptr);
```

Comment:
It will probably better to cast here via 2 `static_cast`.
```
class IMainObject; // Forward declaration
...
return static_cast<Typeto*>(static_cast<IMainObject*>(ptr));
```